shell/services/node/node_service.cc

```diff
@@ -104,7 +108,8 @@ void NodeService::Initialize(node::mojom::NodeServiceParamsPtr params) {
         js_env_->DestroyMicrotasksRunner();
         node::Stop(env, node::StopFlags::kDoNotTerminateIsolate);
         node_env_stopped_ = true;
-        receiver_.ResetWithReason(exit_code, "");
+        std::move(termination_callback_).Run(exit_code);
```

Comment:
What is wrong with using the previous disconnection handler to signal the exit code in this case ?

Reply:
@deepak1556 if we add the previous disconnect handler we end up with the same issue - that it might erroneously exit with the wrong code.

Reply:
Done in a4ef4e6!